core/src/main/java/software/amazon/awssdk/core/runtime/transform/StandardErrorUnmarshaller.java

```diff
@@ -100,40 +101,54 @@ public String getErrorPropertyPath(String property) {
         return "ErrorResponse/Error/" + property;
     }
 
-    public AmazonServiceException standardErrorPathException(String errorCode, Node in, XPath xpath) throws Exception {
+    public SdkServiceException standardErrorPathException(String errorCode, Node in, XPath xpath) throws Exception {
 
         String errorType = asString("ErrorResponse/Error/Type", in, xpath);
         String requestId = asString("ErrorResponse/RequestId", in, xpath);
         String message = asString("ErrorResponse/Error/Message", in, xpath);
 
-        AmazonServiceException ase = newException(message);
-        ase.setErrorCode(errorCode);
-        ase.setRequestId(requestId);
+        SdkServiceException exception = newException(message);
+        exception.errorCode(errorCode);
+        exception.requestId(requestId);
+        exception.errorType(getErrorType(errorType));
 
-        if (errorType == null) {
-            ase.setErrorType(AmazonServiceException.ErrorType.Unknown);
-        } else if (errorType.equalsIgnoreCase("Receiver")) {
-            ase.setErrorType(AmazonServiceException.ErrorType.Service);
-        } else if (errorType.equalsIgnoreCase("Sender")) {
-            ase.setErrorType(AmazonServiceException.ErrorType.Client);
-        }
+        return exception;
+    }
 
-        return ase;
+    /**
+     * Query/Xml services optionally return an string error type that can is either "Sender" or
+     * "Receiver". These error types correspond to who was identified to be at fault with a
+     * request that caused an exception to be returned.
+     *
+     * Receiver will return {@link ErrorType#SERVICE}. Sender will return {@link ErrorType#CLIENT}.
+     * All other values will return {@link ErrorType#UNKNOWN}.
+     *
+     * @param errorType - String error type from returned response.
+     * @return {@link ErrorType}
+     */
+    public ErrorType getErrorType(String errorType) {
+        if ("Receiver".equals(errorType)) {
+            return ErrorType.SERVICE;
+        } else if ("Sender".equals(errorType)) {
+            return ErrorType.CLIENT;
+        } else {
+            return ErrorType.fromValue(errorType);
+        }
     }
 
     @ReviewBeforeRelease("We shouldn't have S3 speific code in core. Also the way this is doesn't" +
```

Comment:
Do we still need this with Zoe's change? Or has that not been merged yet?

Reply:
it got removed in my PR, which got merged after this. so that's old code you are seeing 🙂